ucp/_libs/ucp_py.pyx

```diff
@@ -735,6 +776,12 @@ async def get_endpoint(peer_ip, peer_port, timeout=None):
 
     ep = Endpoint()
     connection_established = False
+    ipr = IPRoute()
+    route = ipr.route('get', dst=peer_ip.decode())
+    my_ip = None
+    for attr in route[0]['attrs']:
+        if attr[0] == 'RTA_PREFSRC':
```

Comment:
@mrocklin I'm not sure if this is standard across platforms but it worked on the machine I tried